### ISSUE TITLE
HARDVS dataset question

### ISSUE THREAD

**Issue by junkangfang** [jiazhou-garland/ExACT#2]
I want to know that can i use the Event Images provided by HARDVS, without having to transform the raw events into event frames？

**Comment by jiazhou-garland** [comment]
Sure, but performance should decrease as the model is trained based on the proposed AFE representation.

**Comment by caomq123** [comment]
Could you please tell me why I am able to reproduce the performance normally on the SeAct and PAF datasets, but the performance on the HARDVS dataset is always below 2.5%? Have I overlooked some important information?

**Comment by jiazhou-garland** [comment]
Hello, please provide more your experiment settings on the HARDVS dataset.

**Comment by caomq123** [comment]
Experiments settings as follows:
Trainer:
  exp_group_name: HARDVS
  exp_num: 1e-5->1e-6 b24
  GPU_ids: [0]
  lr: 1e-5
  min_lr: 1e-6
  weight_decay: 0.05
  epoch: 25
  accumulation_steps: 1 # gradient accumulation, if not 0, remember lr *= accumulation_steps
  print_freq: 1
  Precision: fp16 # CLIP model default is float16
  seed: 101 # 3407 42 114514

MODEL:
  Load_Path: "Path-to-/HARDVS-[90.10].bin" # TODO Change to your directory
  BACKBONE:
    Name: "ViT-B/16"
    Path:
    # CLIP model default is float16
    PRE_ENCODING: "fp16"
    #Download path to PRE_trained_model
    PRE_trained_model: 'ViT-B-16.pt' # TODO Change to your directory
  EventEncoder:
    train_clip_backbone: True
    # feature output from the No.3 ViT block, -1 denotes output whithout the low-level ferature
    use_cross_frame_prompts: True
    use_event_modality_prompts: True
    num_event_modality_prompts: 16
    use_temporal_encoding: True
    Low_level_feature_idx: [1]
  TextEncoder:
    init_ctx: True
    CTX_INIT: "A series of photos recording human action for" #"A series of photos recording human action for" # A sketch image of a
    # length of text prompts
    leranable_ctx: True
    N_CTX: 16
    use_event_bias_textual_prompts: False
  Conceptualizing_Alignment:
    num_concept: 1  # 32 16  8  4   2
    concept_dim: 512 # 16 32 64 128 256
    sample_num: 5

Dataset:
  Input_size: [224,224]
  Train:
    Path: "ExACT/Dataloader/HARDVS/HARDVS_sampled_train_6_3_1.txt" # TODO Change to your directory
    Batch_size: 8 # 24
    Representation: 'gray_scale' # mlp_learned, frame, gray_scale, rgb
    Agumentation: False
  Val:
    Path: "ExACT/Dataloader/HARDVS/HARDVS_sampled_test_6_3_1.txt"
    Batch_size: 1 # 24
    Representation: 'gray_scale' # mlp_learned, frame, gray_scale, rgb
    Agumentation: False
  Classnames: "ExACT/Dataloader/HARDVS/HARDVS_300_class.json" # TODO Change to your directory

LossFunction:
  loss_sigma_threshold: 15000
  loss_cross_entroy: False
  loss_cross_entroy_ori: False
  loss_logsigma_sum: True
  loss_SmoothL1Loss_txt: True
  loss_SmoothL1Loss_ev: True
self.sample_event_num_min = 150000
self.sample_event_threshold = 40

Results as follows:
![ev](https://github.com/user-attachments/assets/269cf3aa-067d-43e4-b2d3-71bfd59ed1b9)
![ev_ori](https://github.com/user-attachments/assets/8acb31dc-734e-4253-95ec-5c52d121e383)
And the "train_dp_HARDVS" file has not been changed.
Thank you for your patient response.

**Comment by jiazhou-garland** [comment]
Hello, do you use the **raw events** of the HARDVS dataset (rather than the released event frames) and then preprocess with our AFE representation?

**Comment by caomq123** [comment]
I use the raw events and then preprocess with AFE representation:
![image](https://github.com/user-attachments/assets/4ded23e4-96be-4338-8852-c8fea2715264)
![image](https://github.com/user-attachments/assets/3ab1514b-318d-470f-b195-515c0e354ca1)

**Comment by jiazhou-garland** [comment]
I have updated the train code of the HARDVS dataset. Try to use the updated one.
Besides, there few points you may check if the updated train code still does not work:

1. Use the training code of other datasets;
2. Use part of the dataset for mico-training;
3. Load the released pre-trained weights on the HARDVS dataset and train several epochs to see what will happen (lower the lr);
4. Ensure the preprocessed dataset(event frames) are rationale;

**Comment by caomq123** [comment]
I'll give it a try, thank you.

**Comment by WJ-self** [comment]
same low accuracy on HARDVS dataset problem here🤔

**Comment by Zizzzzzzz** [comment]
> Could you please tell me why I am able to reproduce the performance normally on the SeAct and PAF datasets, but the performance on the HARDVS dataset is always below 2.5%? Have I overlooked some important information?

When reproducing the results on the SeACT dataset, how did you set the parameters self.sample_event_num_min and self.sample_event_threshold?

**Comment by Zizzzzzzz** [comment]
> same low accuracy on HARDVS dataset problem here🤔

Have you resolved this issue? The accuracy I achieved during reproduction is around 60%.
